miden-tx/src/executor/mod.rs

```diff
@@ -144,12 +141,16 @@ impl TransactionExecutor {
         // load note script MAST into the MAST store
         self.mast_store.load_transaction_code(&tx_inputs, &tx_args);
 
+        for code in self.account_codes.values() {
+            self.mast_store.load_account_code(code);
+        }
```

Comment:
I'm still not quite following why we need this code. I think it can be safely removed, no?

Reply:
Indeed, sorry for the confusion